### PR TITLE
splice: first pass on splice_out command

### PR DESCRIPTION
**Pull request by ddustin**
Added splice_out RPC command and a simple test of it as well.

### CODE REVIEW

**Comment by ddustin** [comment]
Should use `wallet_utxo_boost` from PR https://github.com/ElementsProject/lightning/pull/6780

File: tests/test_splicing.py [hunk 1]
```diff
@@ -171,6 +171,30 @@ def test_splice_out(node_factory, bitcoind):
...
 
 
+@pytest.mark.openchannel('v1')
+@pytest.mark.openchannel('v2')
+@unittest.skipIf(TEST_NETWORK != 'regtest', 'elementsd doesnt yet support PSBT features we need')
+def test_splice_out_ez(node_factory, bitcoind):
+    l1, l2 = node_factory.line_graph(2, fundamount=1000000, wait_for_announce=True, opts={'experimental-splicing': None})
+
+    l1.rpc.splice_out(l1.get_channel_id(l2), 100000)
+
```
Comment (niftynei) [inline tests/test_splicing.py:181]: nit: check that the money moved!

File: lightningd/channel_control.c [hunk 2]
```diff
@@ -293,10 +325,16 @@ static void handle_splice_confirmed_init(struct lightningd *ld,
 		return;
 	}
 
-	struct json_stream *response = json_stream_success(cc->cmd);
-	json_add_string(response, "psbt", psbt_to_b64(tmpctx, psbt));
+	if (cc->auto_complete) {
+		subd_send_msg(channel->owner,
+			      take(towire_channeld_splice_update(NULL, psbt)));
+	}
```
Comment (niftynei) [inline lightningd/channel_control.c:331]: nit: less vertical space please

Comment (niftynei) [reply lightningd/channel_control.c:331]: actually let's regroup about this and talk about how to make it better!

File: lightningd/channel_control.c [hunk 3]
```diff
@@ -2079,6 +2104,122 @@ static const struct json_command splice_signed_command = {
...
+	if (!b32script) {
+		return command_fail(cmd, LIGHTNINGD,
+				    "Failed to generate change address."
+				    " Keys generation failure");
+	}
+	txfilter_add_scriptpubkey(cmd->ld->owned_txfilter, b32script);
+
+	psbt_append_output(psbt, b32script, *amount);
+
+	weight = wally_psbt_weight(psbt);
```
Comment (niftynei) [inline lightningd/channel_control.c:2197]: i think you're going to want to add the signature witness byte weight as well, or at least an estimate for it.

**Review by niftynei — COMMENTED** [review]
This needs to be one level up, in a plugin. Let's circle up about it soon and get moving that direction instead.

Nice and short tho!

**Comment by ddustin** [comment]
Working on a generic splice json / script protocol that can handle doing any kind of splice.

Then splice-out, splice-in, etc will use that RPC and pass it the correct json to do it.

**Pull request closed without merging**